### PR TITLE
Ws connection middleware for redux

### LINKED ISSUE #19: Admin UI: link websocket for receiving messages to the store
Definition of done:
 - work to be done only in Redux or app, no html
 - each message should be stored in redux
 - each message should have a property 'seen' and 'timestamp'
 - there should be a counter of unseen messages in redux

### PR DESCRIPTION
**Pull request by diegoalzate**
closes #19 

### How does this work? 
[Redux middleware](https://redux.js.org/understanding/history-and-design/middleware#the-final-approach) it provides a third-party extension point between dispatching an action, and the moment it reaches the reducer. So when a initializeWebsocket action is triggered it will first run run through the middleware creating a websocket connection before reaching the actual reducer.

### Why?
The WebSocket instance is not serializable and thus technically not something that should be stored in Redux state. So there were 2 alternatives considered in this pr:

#### [rtk query streaming updates](https://redux-toolkit.js.org/rtk-query/usage/streaming-updates)
With this solution we could have an easy way to mount ws connection and tear it down, however there are a lot of concepts the team would need to learn and it also provides less customization. 

#### custom redux middleware
Even though this solution is more code, the concept of middleware is easier to understand and provides us a solution that can adapter better to any changes that may happen on hoprd side.

### why was LocalRootState created?
The middleware has access to the whole store, however I could not reference the store type inside the actual middleware because the middleware will be used in the store. That would create a circular dependency. We could create the RootStore type by hand instead of inferring but I think the store is more likely to change than the middleware.

### How can this be better?
- add common ws logic to hopr-sdk and use it here

### CODE REVIEW

**Comment by diegoalzate** [comment]
blocked by https://github.com/hoprnet/hopr-sdk/pull/56

**Comment by mjadach-iv** [comment]
Lets wait with merge until older PRs are merged.

**Review by mjadach-iv — CHANGES_REQUESTED** [review]
Rebase needed

File: src/store/slices/node/websocketMiddleware.ts
Comment (mjadach-iv) [inline src/store/slices/node/websocketMiddleware.ts:33]: shouldn't this part be provided by the HOPR-SDK?

File: src/store/slices/node/websocketMiddleware.ts
Comment (mjadach-iv) [inline src/store/slices/node/websocketMiddleware.ts:38]: We should update the WebsocketHelper to comly with the apiEndpoint and apiToken standard we have in all the other calls, and then use it here

**Review by mjadach-iv — APPROVED** [review]

**Pull request merged**
